File: java/state/CancelledOrderState.java
package state;

public class CancelledOrderState implements OrderState {
    private final OrderManagement orderManagement;

    public CancelledOrderState(OrderManagement orderManagement) {
        this.orderManagement = orderManagement;
    }

    @Override
    public void processOrder() {
        System.out.println("Cannot process the order at the current state...");
    }

    @Override
    public void shipOrder() {
        System.out.println("Cannot ship the order at the current state...");
    }

    @Override
    public void deliverOrder() {
        System.out.println("Cannot deliver the order at the current state...");
    }

    @Override
    public void cancelOrder() {
        System.out.println("Order is already cancelled!");
    }
}

File: PHP/Patterns/src/Behavioral/State/CancelledOrderState.php
<?php

namespace src\Behavioral\State;

class CancelledOrderState implements OrderState
{
    private OrderManagement $orderManagement;

    /**
     * @param OrderManagement $orderManagement
     */
    public function __construct(OrderManagement $orderManagement)
    {
        $this->orderManagement = $orderManagement;
    }

    function processOrder()
    {
        print_r("Can't process the order at the current state \n");
    }

    function shipOrder()
    {
        print_r("Can't ship the order at the current state \n");
    }

    function deliverOrder()
    {
        print_r("Can't deliver the order at the current state \n");
    }

    function cancelOrder()
    {
        print_r("This order is already cancelled \n");

    }
}